diff --git a/Users/grotter/Desktop/tfa_basic/includes/tfa_sms.inc b/tfa_basic/includes/tfa_sms.inc
index 27861e2d..8eb70c94 100644
--- a/Users/grotter/Desktop/tfa_basic/includes/tfa_sms.inc
+++ b/tfa_basic/includes/tfa_sms.inc
@@ -224,4 +224,18 @@ class TfaBasicSmsSetup extends TfaBasicSms implements TfaSetupPluginInterface {
     return TRUE;
   }
 
+  /**
+  * {@inheritdoc}
+  */
+  public function ready() {
+    // We can only send a code if there is a phone number.
+    if (!empty($this->mobileNumber)) {
+      // We should only send codes if TFA is enabled for this account.
+      $user_data = tfa_basic_get_tfa_data(user_load($this->context['uid']));
+      if (!empty($user_data['status'])) {
+        return TRUE;
+      }
+    }
+    return FALSE;
+  }
 }
diff --git a/Users/grotter/Desktop/tfa_basic/tfa_basic.module b/tfa_basic/tfa_basic.module
index af6254e0..3d3ba0f1 100644
--- a/Users/grotter/Desktop/tfa_basic/tfa_basic.module
+++ b/tfa_basic/tfa_basic.module
@@ -266,7 +266,7 @@ function tfa_basic_tfa_context_alter(&$context) {
       }
     }
     // Remove SMS from validation if set.
-    if ('tfa_basic_sms' == $context['plugins']['validate']) {
+    if ('tfa_basic_sms' == $context['plugins']['validate'] && !empty($fallback_plugins)) {
       $context['plugins']['validate'] = array_shift($fallback_plugins);
     }
     $context['plugins']['fallback'] = $fallback_plugins;
